enterprise/server/remote_execution/snaputil/snaputil.go

```diff
@@ -92,15 +96,23 @@ func GetArtifact(ctx context.Context, localCache interfaces.FileCache, bsClient
 		return 0, err
 	}
 	defer f.Close()
+
+	// If the proxy is enabled, snapshots are not saved to the remote cache to minimize
+	// high network transfer. Snapshots can't be shared across different machine
+	// types, so there's no reason to support snapshot sharing across clusters.
+	ctx = metadata.AppendToOutgoingContext(ctx, proxy_util.SkipRemoteKey, "true")
```

Comment:
if it were me i'd add `SetSkipRemote(...) context` to `proxy_util`, but whichever you like better.

Reply:
Done